### PR TITLE
Feature/toggle api visibility

### LINKED ISSUE #1020: Setting for API visibility
Add a setting / option for API owner so that they can hide their API backend from catalog. This is one of the first steps from supporting not only open APIs but also closed APIs.

_Open issue: Should this setting be available in API profile view or in Edit API settings view?_

Related issue: #1207

# User story
> As an API owner 
I want to be able to select whether my API is publicly visible in the catalog or not

# Definition of done
* [ ] API owner has an option to select API visibility in catalog: visible or hidden (UI widget)
* [x] The API owner and Apinf administator(s) can easily see what is the current/selected visibility setting of API 
* [x] If API is hidden by owner, only administrators and the particular owner can see it in the catalog. It is hidden for all other users.
* [x] Not authorized users cannot view  the data: they will get a not authorized message

# Wireframe

apivisibilityvisible

apivisibilityhidden

### PR DESCRIPTION
**Pull request by jykae**
Closes #1020 

## Proposed changes

- add Bootstrap Switch library
- add visibility switch component on API view

### Additional cleanup

- Remove unused "packages.json"

### CODE REVIEW

**Comment by jykae** [comment]
Addded WIP because, permission checks not added yet.

**Comment by jykae** [comment]
Currently showing on "API view" page, better place to be decided.

nayttokuva 2016-07-22 kello 15 08 36

**Comment by jykae** [comment]
## Self-reminder for next week: 
- consider doing view rights check on publication level (no need to fetch APIs that won't be visible for current user).
- add view rights check on "API view" page -> show "not authorized" message

**Comment by jykae** [comment]
@apinf/developers please review

File: apis/client/latest_api/latestApi.html [hunk 1]
```diff
@@ -6,18 +6,20 @@ <h2 class="feature-section-heading">{{_ "latestApiBackends_Heading" }}</h2>
       <hr class="primary">
     </div>
   </div>
-  {{#each latestBackends }}
+  {{#each api in latestBackends }}
+  {{#if api.currentUserCanView }}
   <div class="col-md-3 col-sm-6 hero-feature">
     <div class="thumbnail api-box">
-      <h3><a href="/api/{{_id}}">{{ name }}</a></h3>
+      <h3><a href="/api/{{_id}}">{{ api.name }}</a></h3>
```
Comment (nnn) [inline apis/client/latest_api/latestApi.html]: Links are broken on home page.

**Comment by nnn** [comment]
No visual feedback when changing visibility.

**Comment by nnn** [comment]
By default, let's set all APIs visible.

**Comment by jykae** [comment]
@nnn Taking look at this tomorrow.

**Comment by nnn** [comment]
@jykae, since we are moving all the settings to a separate tab, we need to change the way visibility is toggled.

**Comment by jykae** [comment]
@nnn Ok, what to do with this then? Is there wireframe I should follow?

**Comment by brylie** [comment]
Ping @Nazarah, can you point @jykae to any related wireframe(s)?

**Comment by jykae** [comment]
Got wireframes, closing this when I have taken usable parts of code to new PR.

**Comment by nnn** [comment]
@jykae, wait for @Nazarah updated wireframes or discuss with her before you start.

**Comment by jykae** [comment]
PR #1372 replaces this. Closing..

**Pull request closed without merging**